alica_tests/include/test_alica.h

```diff
@@ -128,22 +129,23 @@ class AlicaTestMultiAgentFixture : public AlicaTestMultiAgentFixtureBase
                 std::make_unique<alica::ConstraintCreator>(), std::make_unique<alica::BehaviourCreator>(),
                 std::make_unique<alica::PlanCreator>());
 
-        spinner = std::make_unique<ros::AsyncSpinner>(4);
-        spinner->start();
         for (int i = 0; i < getAgentCount(); ++i) {
             alica::AlicaContext* ac = new alica::AlicaContext(
                     alica::AlicaContextParams(getHostName(i), path + "/etc/", getRoleSetName(),
                                               getMasterPlanName(), stepEngine()));
             ASSERT_TRUE(ac->isValid());
+            spinners.emplace_back(std::make_unique<ros::AsyncSpinner>(4));
```

Comment:
its useless to have multiple spinners if they are processing from same global callback queue. You can have a single spinner or otherwise have 1 cb queue per spinner and pass respective queue in AlicaRosCommunication and Timer Factory constructor. Also i don't see the spinner->stop calls before ac->terminate() in tearDown